comet/mutation_data.py

```diff
@@ -20,31 +20,49 @@ def load_mutation_data(filename, patientFile=None, geneFile=None, minFreq=0):
       * **patients** (*list*) - patients in the mutation data.
       * **geneToCases** (*dictionary*) - mapping of genes to the patients in which they are mutated.
       * **patientToGenes** (*dictionary*) - mapping of patients to the genes they have mutated.
+      * **subtypes** (*list*) - list of unique subtypes.
     """
-    # Load the whitelists (if applicable)
+    # Load the patient whitelist (if applicable)
     if patientFile:
         with open(patientFile) as f:
             patients = set( l.rstrip().split()[0] for l in f if not l.startswith("#") )
     else:
         patients = None
 
+    # Load the subtype information (if applicable)
+    from collections import defaultdict
+    subtypeDict = defaultdict( lambda: None, dict() )
+    subtypes = set()
+    if subtypeFile:
+        subtypeDict = defaultdict( lambda: None, dict() )
+        subtypes = set()
+        with open( subtypeFile ) as f:
+            sts = [ l.rstrip().split( '\t' ) for l in f if not l.startswith( '#' ) ]
+            for p, s in sts:
+                subtypeDict[p] = s
+                subtypes.add( s )
+
+    # Load the gene whitelist (if applicable)
     if geneFile:
         with open(geneFile) as f:
             genes = set( l.rstrip().split()[0] for l in f if not l.startswith("#") )
+        genes |= subtypes
     else:
         genes = set()
 
     # Parse the mutation matrix
-    from collections import defaultdict
     geneToCases, patientToGenes = defaultdict(set), defaultdict(set)
     with open(filename) as f:
         arrs = [ l.rstrip().split("\t") for l in f if not l.startswith("#") ]
         for arr in arrs:
             patient, mutations = arr[0], set(arr[1:])
 
+            if subtypeDict[patient]:
+                mutations = mutations.union( subtypes.difference( set( [subtypeDict[patient]] ) ) )
```

Comment:
I would write something like the following instead:

    mutations |= set( s for s in subtypes if subtypeDict[patient] != s )